src/parse.y

```diff
@@ -1265,45 +1288,49 @@ type:
 				$$ = new VectorType({AdoptRef{}, $3});
 				}
 
-	|	TOK_FUNCTION func_params
+	|	TOK_FILE TOK_OF type
 				{
-				set_location(@1, @2);
-				$$ = $2;
+				set_location(@1, @3);
+				$$ = new FileType({AdoptRef{}, $3});
 				}
 
-	|	TOK_EVENT '(' formal_args ')'
+	|	TOK_FILE
 				{
-				set_location(@1, @3);
-				$$ = new FuncType({AdoptRef{}, $3}, nullptr, FUNC_FLAVOR_EVENT);
+				set_location(@1);
+				$$ = new FileType(base_type(TYPE_STRING));
 				}
 
-	|	TOK_HOOK '(' formal_args ')'
+	|	TOK_OPAQUE TOK_OF TOK_ID
 				{
 				set_location(@1, @3);
-				$$ = new FuncType({AdoptRef{}, $3}, base_type(TYPE_BOOL), FUNC_FLAVOR_HOOK);
+				$$ = new OpaqueType($3);
 				}
 
-	|	TOK_FILE TOK_OF type
+type:
+		simple_type
+	|	TOK_FUNCTION func_params
 				{
-				set_location(@1, @3);
-				$$ = new FileType({AdoptRef{}, $3});
+				set_location(@1, @2);
+				$$ = $2;
 				}
 
-	|	TOK_FILE
+	|	TOK_HOOK '(' formal_args ')'
 				{
-				set_location(@1);
-				$$ = new FileType(base_type(TYPE_STRING));
+				set_location(@1, @3);
+				$$ = new FuncType({AdoptRef{}, $3}, base_type(TYPE_BOOL), FUNC_FLAVOR_HOOK);
 				}
 
-	|	TOK_OPAQUE TOK_OF TOK_ID
+	|	TOK_EVENT '(' formal_args ')'
 				{
 				set_location(@1, @3);
-				$$ = new OpaqueType($3);
+				$$ = new FuncType({AdoptRef{}, $3}, nullptr, FUNC_FLAVOR_EVENT);
 				}
 
 	|	resolve_id
 			{
-			if ( ! $1 || ! ($$ = $1->IsType() ? $1->GetType().get() : nullptr) )
+			if ( $1 && $1->GetType()->Tag() == TYPE_TYPE )
+				$$ = $1->GetType()->AsTypeType()->GetType()->Ref();
```

Comment:
Note that we can remove this, then the only thing that this adds is types passing in to BIFs (so it lets you do `type_name(string)` but not `local ty = string;`).

That may be better if it's a `ConstExpr` since this line kind of conflates a dynamic type value with a statically known type

So I'm okay removing this just to make the new `ConstExpr` make sense

Reply:
> Note that we can remove this, then the only thing that this adds is types passing in to BIFs (so it lets you do type_name(string) but not local ty = string;).

I'm trying to think of when you would ever do this in a Zeek script, but C++ allows type-aliasing like this at the function level too with `using` statements. Not quite the same as assigning a type to a variable though.